soc/arm/nxp_imx/rt6xx/soc.c

```diff
@@ -0,0 +1,216 @@
+	.audio_pll_mult = kCLOCK_AudioPllMult22  /* Divide by 22 */
+};
+#endif
+
+#ifdef CONFIG_NXP_IMX_RT_BOOT_HEADER
+extern char z_main_stack[];
+extern char _flash_used[];
+
+extern void z_arm_reset(void);
+extern void z_arm_nmi(void);
+extern void z_arm_hard_fault(void);
+extern void z_arm_mpu_fault(void);
+extern void z_arm_bus_fault(void);
+extern void z_arm_usage_fault(void);
+extern void z_arm_secure_fault(void);
+extern void z_arm_svc(void);
+extern void z_arm_debug_monitor(void);
+extern void z_arm_pendsv(void);
+extern void z_clock_isr(void);
+extern void z_arm_exc_spurious(void);
+
+__imx_boot_ivt_section void (* const image_vector_table[])(void)  = {
+	(void (*)())(z_main_stack + CONFIG_MAIN_STACK_SIZE),  /* 0x00 */
+	z_arm_reset,							/* 0x04 */
+	z_arm_nmi,								/* 0x08 */
+	z_arm_hard_fault,						/* 0x0C */
+	z_arm_mpu_fault,						/* 0x10 */
+	z_arm_bus_fault,						/* 0x14 */
+	z_arm_usage_fault,						/* 0x18 */
+#if defined(CONFIG_ARM_SECURE_FIRMWARE)
+	z_arm_secure_fault,						/* 0x1C */
+#else
+	z_arm_exc_spurious,
+#endif /* CONFIG_ARM_SECURE_FIRMWARE */
+	(void (*)())_flash_used,				/* 0x20, RT600 image header: imageLength. */
+	0,										/* 0x24, RT600 image header: imageType (Plain Image). */
+	0,										/* 0x28, RT600 image header: authBlockOffset/crcChecksum. */
+	z_arm_svc,								/* 0x2C */
+	z_arm_debug_monitor,					/* 0x30 */
+	(void (*)())image_vector_table,			/* 0x34, RT600 image header: imageLoadAddress. */
+	z_arm_pendsv,							/* 0x38 */
+#if defined(CONFIG_SYS_CLOCK_EXISTS)
+	z_clock_isr,							/* 0x3C */
+#else
+	z_arm_exc_spurious,
+#endif
+};
+#endif /* CONFIG_NXP_IMX_RT_BOOT_HEADER */
+
+/**
+ *
+ * @brief Initialize the system clock
+ *
+ * @return N/A
+ *
+ */
+
+static ALWAYS_INLINE void clock_init(void)
+{
+#ifdef CONFIG_SOC_MIMXRT685S_CM33
+	/* Configure LPOSC clock*/
+	if ((SYSCTL0->PDRUNCFG0 & SYSCTL0_PDRUNCFG0_LPOSC_PD_MASK) != 0) {
+		POWER_DisablePD(kPDRUNCFG_PD_LPOSC); /* Power on LPOSC (1MHz) */
+	}
+
+	/* Configure FFRO clock */
+	if ((SYSCTL0->PDRUNCFG0 & SYSCTL0_PDRUNCFG0_FFRO_PD_MASK) != 0) {
+		POWER_DisablePD(kPDRUNCFG_PD_FFRO);  /* Power on FFRO (48/60MHz) */
+		CLOCK_EnableFfroClk(kCLOCK_Ffro48M); /* Enable FFRO clock*/
+	}
+	if ((SYSCTL0->PDRUNCFG0 & SYSCTL0_PDRUNCFG0_SFRO_PD_MASK) != 0) {
+		/* Configure SFRO clock */
+		POWER_DisablePD(kPDRUNCFG_PD_SFRO); /* Power on SFRO (16MHz) */
+		CLOCK_EnableSfroClk();				/* Wait until SFRO stable */
+	}
+
+	if ((SYSCTL0->PDRUNCFG0 & SYSCTL0_PDRUNCFG0_SYSXTAL_PD_MASK) != 0) {
+		/* Configure SYSOSC clock source */
+		POWER_DisablePD(kPDRUNCFG_PD_SYSXTAL);						 /* Power on SYSXTAL */
+		CLOCK_EnableSysOscClk(true, true, BOARD_SYSOSC_SETTLING_US); /* Enable system OSC */
+	}
+	/* Updated XTAL oscillator settling time */
+	POWER_UpdateOscSettlingTime(BOARD_SYSOSC_SETTLING_US);
+	CLOCK_SetXtalFreq(BOARD_XTAL_SYS_CLK_HZ);			   /* Sets external XTAL OSC freq */
```

Comment:
Please remove the trailing comments in this section. They cause a lot of checkpatch warnings and don't provide much information.